### PR TITLE
[pytorch-finetuning] Introduction of AI Halo Reference Platform

### PR DESCRIPTION
**Pull request by ldokovic-personal**
### **Improvements:**
- Added AI Halo as a device option, supporting both Windows and Linux operating systems.
- Enhanced the user interface on the website to provide a clear overview when users select categories.
- Fixed typo on Model's directory output to match expected title for the 'saving' step.
- Updated the README to include guidelines for formatting datasets, ensuring compatibility with JSON, CSV, and Hugging Face formats.
- Emphasized the importance of maintaining an instruction-response pair in datasets and provided instructions for modifying the format_instruction() function to accommodate custom keys.

_**Edit: This issue down below, has been resolved**_

The check_os_kernel() issue is resolved with the new 3.0.1 image release. The playbook's methods have been reorganized and fully validated for accuracy. Bitsandbytes didn't affected any train method run, even though we covered it in via Notes in [#244](https://github.com/amd/playbooks/pull/244)

All addressed in [Finetune LLMs with Pytorch playbook Report ](https://amdcloud-my.sharepoint.com/:w:/g/personal/ldokovic_amd_com/IQCT7CBYVem9QaZDzJUxLFbrAUpkBcdxruhBSSM4CU889r0?e=fVfEyR)

### CODE REVIEW

File: playbooks/supplemental/pytorch-finetuning/README.md [hunk 1]
```diff
@@ -23,19 +23,50 @@ This tutorial provides step-by-step examples for fine-tuning a large language mo
...
+<!-- @device:end -->
+<!-- @setup:id=activate-venv command="source venv/bin/activate" -->
+<!-- @os:end -->
+
+<!-- @os:windows -->
+<!-- @device:halo_box -->
+<!-- @test:id=create-venv timeout=60 -->
+```powershell
+python -m venv venv --system-site-packages
+venv\Scripts\activate.bat
```
Comment (sreeram-11) [inline playbooks/supplemental/pytorch-finetuning/README.md]: `venv\Scripts\activate.bat` won’t work in PowerShell since it’s a CMD script.

Ideally,
- In PowerShell → use: `.\venv\Scripts\Activate.ps1`
- In CMD → use: `venv\Scripts\activate.bat`

File: playbooks/supplemental/pytorch-finetuning/README.md [hunk 2]
```diff
@@ -23,19 +23,50 @@ This tutorial provides step-by-step examples for fine-tuning a large language mo
...
+```
+<!-- @test:end -->
+<!-- @device:end -->
+<!-- @setup:id=activate-venv command="source venv/bin/activate" -->
+<!-- @os:end -->
+
+<!-- @os:windows -->
+<!-- @device:halo_box -->
+<!-- @test:id=create-venv timeout=60 -->
+```powershell
```
Comment (sreeram-11) [inline playbooks/supplemental/pytorch-finetuning/README.md:46]:
```suggestion
```bash
```

or


```suggestion
```cmd
```

File: playbooks/supplemental/pytorch-finetuning/README.md [hunk 3]
```diff
@@ -23,19 +23,50 @@ This tutorial provides step-by-step examples for fine-tuning a large language mo
...
+<!-- @os:end -->
+
 
 <!-- @os:windows -->
+<!-- @device:halo,stx,krk,rx7900xt,rx9070xt -->
 <!-- @test:id=create-venv timeout=60 -->
-```cmd
+```powershell
 python -m venv venv
 venv\Scripts\activate.bat
```
Comment (sreeram-11) [inline playbooks/supplemental/pytorch-finetuning/README.md]: Again, for PowerShell, use `.\venv\Scripts\Activate.ps1`

File: playbooks/supplemental/pytorch-finetuning/README.md [hunk 4]
```diff
@@ -23,19 +23,50 @@ This tutorial provides step-by-step examples for fine-tuning a large language mo
...
 ```
 <!-- @test:end -->
+<!-- @device:end -->
 <!-- @setup:id=activate-venv command="venv\Scripts\activate.bat" -->
```
Comment (sreeram-11) [inline playbooks/supplemental/pytorch-finetuning/README.md]: If using PowerShell, update the following:


```suggestion
<!-- @setup:id=activate-venv command=".\venv\Scripts\Activate.ps1" -->
```

File: playbooks/supplemental/pytorch-finetuning/README.md [hunk 5]
```diff
@@ -23,19 +23,50 @@ This tutorial provides step-by-step examples for fine-tuning a large language mo
...
+<!-- @os:windows -->
+<!-- @device:halo_box -->
+<!-- @test:id=create-venv timeout=60 -->
+```powershell
+python -m venv venv --system-site-packages
+venv\Scripts\activate.bat
+```
+<!-- @test:end -->
+<!-- @device:end -->
+<!-- @setup:id=activate-venv command="source venv/bin/activate" -->
```
Comment (sreeram-11) [inline playbooks/supplemental/pytorch-finetuning/README.md]: This is `bash` command used in Linux. Please use the below suggestion if using PowerShell:

```suggestion
<!-- @setup:id=activate-venv command=".\venv\Scripts\Activate.ps1" -->
```

File: playbooks/supplemental/pytorch-finetuning/README.md [hunk 6]
```diff
@@ -44,32 +75,54 @@ python3 -m venv venv
...
+<!-- @device:end -->
 <!-- @setup:id=activate-venv command="source venv/bin/activate" -->
 <!-- @os:end -->
 
-### Installing Basic Dependencies
+#### Installing Basic Dependencies
 <!-- @require:pytorch -->
 
-### Additional Dependencies
+<!-- @os:linux -->
```
Comment (sreeram-11) [inline playbooks/supplemental/pytorch-finetuning/README.md]: Please remove lines 85–100, they’re not needed.

The `@require` tags for the Halo box already show up under the heading “Already pre-installed on your AMD Halo Developer Platform”, so this section is redundant.

File: playbooks/supplemental/pytorch-finetuning/README.md [hunk 7]
```diff
@@ -44,32 +75,54 @@ python3 -m venv venv
...
+##### **HaloBox: Skip driver and PyTorch Dependecies**
+No need to set up driver and PyTorch dependencies, as these configurations are pre-installed.
+<!-- @test:end -->
+<!-- @device:end -->
+<!-- @os:end -->
+
+#### Additional Dependencies
 
 <!-- @os:linux -->
+<!-- @device:halo_box,halo,stx,krk,rx7900xt,rx9070xt -->
```
Comment (sreeram-11) [inline playbooks/supplemental/pytorch-finetuning/README.md]: It’s not needed since these are already the default target platforms, and the code below will run on them even if the line isn’t explicitly specified.

```suggestion

```

File: playbooks/supplemental/pytorch-finetuning/README.md [hunk 8]
```diff
@@ -44,32 +75,54 @@ python3 -m venv venv
...
 <!-- @test:id=install-deps timeout=300 setup=activate-venv -->
 ```bash
 pip install transformers==4.57.1 safetensors==0.6.2 accelerate peft trl bitsandbytes "fsspec[http]>=2023.1.0,<=2025.9.0"
 ```
 <!-- @test:end -->
+<!-- @device:end -->
```
Comment (sreeram-11) [inline playbooks/supplemental/pytorch-finetuning/README.md]:
```suggestion

```

File: playbooks/supplemental/pytorch-finetuning/README.md [hunk 9]
```diff
@@ -44,32 +75,54 @@ python3 -m venv venv
...
 <!-- @os:end -->
 
 <!-- @os:windows -->
+<!-- @device:halo_box,halo,stx,krk,rx7900xt,rx9070xt -->
```
Comment (sreeram-11) [inline playbooks/supplemental/pytorch-finetuning/README.md]:
```suggestion

```

File: playbooks/supplemental/pytorch-finetuning/README.md [hunk 10]
```diff
@@ -44,32 +75,54 @@ python3 -m venv venv
...
 **Windows:** Only core packages are tested and supported here. **bitsandbytes is not well supported on Windows**, so the Windows install omits it; use LoRA or full fine-tuning on Windows (QLoRA requires bitsandbytes and is intended for Linux).
 <!-- @test:id=install-deps timeout=300 setup=activate-venv -->
 ```bash
 pip install transformers==4.57.1 safetensors==0.6.2 datasets==4.2.0 accelerate peft trl "fsspec[http]>=2023.1.0,<=2025.9.0"
 ```
 <!-- @test:end -->
+<!-- @device:end -->
```
Comment (sreeram-11) [inline playbooks/supplemental/pytorch-finetuning/README.md]:
```suggestion

```

File: playbooks/supplemental/pytorch-finetuning/README.md (hunk 1 above)
Comment (adamlam2-amd) [reply playbooks/supplemental/pytorch-finetuning/README.md]: fixed

File: playbooks/supplemental/pytorch-finetuning/README.md (hunk 5 above)
Comment (adamlam2-amd) [reply playbooks/supplemental/pytorch-finetuning/README.md]: should be fixed
